server/settings_test.go

```diff
@@ -281,6 +282,86 @@ func TestSettings(t *testing.T) {
 					assert.Equal(t, "864000s", res.Payload.Settings.DataRetention)
 					assert.Equal(t, []string{"aws", "aws-cn"}, res.Payload.Settings.AWSPartitions)
 				})
+
+				t.Run("Set Alert Manager Rules", func(t *testing.T) {
+					defer teardown(t)
+					body := server.ChangeSettingsBody{
+						AlertManagerAddress: "localhost:1234",
+						AlertManagerRules: `groups:
+- name: example
+  rules:
+  - alert: HighRequestLatency
+    expr: job:request_latency_seconds:mean5m{job="myjob"} > 0.5
+    for: 10m
+    labels:
+      severity: page
+    annotations:
+      summary: High request latency`,
+					}
+
+					res, err := serverClient.Default.Server.ChangeSettings(&server.ChangeSettingsParams{
+						Body:    body,
+						Context: pmmapitests.Context,
+					})
+					require.NoError(t, err)
+					assert.Equal(t, res.Payload.Settings.AlertManagerAddress, body.AlertManagerAddress)
+					assert.Equal(t, res.Payload.Settings.AlertManagerRules, body.AlertManagerRules)
+
+					gets, err := serverClient.Default.Server.GetSettings(nil)
+					require.NoError(t, err)
+					assert.Equal(t, body.AlertManagerAddress, gets.Payload.Settings.AlertManagerAddress)
+					assert.Equal(t, body.AlertManagerRules, gets.Payload.Settings.AlertManagerRules)
+				})
+
+				t.Run("Empty Alert Manager Rules should not remove data", func(t *testing.T) {
```

Comment:
Are you sure that this test always runs after the previous one? All tests should be independent.